wnfs/src/private/node/header.rs

```diff
@@ -130,7 +130,7 @@ impl PrivateNodeHeader {
     /// ```
     /// use std::rc::Rc;
     /// use wnfs::private::{
-    ///     PrivateFile, AesKey,
```

Comment:
Why did this parameter disappear?

Also, if it exists for Rust, I'd strongly suggest setting up doctests.

Reply:
We do run doctests. This disappeared because it's an unused import (and I removed the struct in this PR). Doctests is actually how I noticed that.

Reply:
Oh interesting! Is it just testing that it compiles, or is there an implicit assertion of some kind?

Reply:
It runs the code. This doctest in particular doesn't have an assertion, apart from the fact that if anything in here we're to panic, the doctest would fail, but others do.

Reply:
Right gotcha — that's the same behaviour in Elixir (it catches any panics and just fails the doctest).

Maybe there are different pattenrs in Rust than Elixir, and it's your code so this is just a suggestion from a non-Rustaciean: the type system will already catch the kind of thing that's written here, so while the documentation side is helpful, it may be _more_ helpful with some broader exploration. I've found that doctests can do a really great job of encouraging people to write great docs when they have to explain why you'd care about the function and/or its output, and often turns into stories about "you care about X, so this gives you back Y which you need for Z".

Reply:
Agreed. The docs could use some work.
I'd actually have preferred if we had *less* doctest while the API is still in flux, and then do a doc pass when it's more stable. It's been a lot of work updating doctests for changes previously.